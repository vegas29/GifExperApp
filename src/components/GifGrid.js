// import { useState, useEffect } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
import PropTypes from 'prop-types';

const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    console.log(images);

    return (
        <>
            <h2 className="text-white text-center py-5 text-4xl md:text-3xl uppercase animate__animated animate__fadeIn">{category}</h2>

            {loading && 
            <div className="spinner">
                <div className="cube1"></div>
                <div className="cube2"></div>
            </div>
            }

            <div className="card-grid">
        
                { images.map(img => 
                    <GifGridItem 
                        key={img.id} 
                        {...img}
                    />
                )}

            </div>
        </>
        
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
 
export default GifGrid;