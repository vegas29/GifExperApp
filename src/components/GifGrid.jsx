import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
            <h2 className="text-white text-center py-5 text-4xl md:text-3xl uppercase animate__animated animate__fadeIn">{ category }</h2>

            {/* { isLoading && 
                <span class="loader"></span>
            } */}

            {/* Test Loader */}

            {
                isLoading && ( <h2>Cargando...</h2> )
            }

            <div className="card-grid">
        
                { images.map(img => 
                    <GifGridItem 
                        key={img.id} 
                        {...img}
                    />
                )}

            </div>
        </>
        
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
 