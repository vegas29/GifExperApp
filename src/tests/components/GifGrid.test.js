import React from 'react';
import {shallow} from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Probando el componente GifGrid', () =>{
    const category = 'hola mundo';
    test('Debe renderizar el componente correctamente', ()=>{
        useFetchGifs.mockReturnValue({
            data: [],
            loading:true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', ()=>{

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa/hola.jpg',
            title: 'Cualquier cosa'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false); 
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
        

    })
})