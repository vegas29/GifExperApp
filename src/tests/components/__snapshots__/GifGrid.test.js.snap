// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente GifGrid Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h2
    className="text-white text-center py-5 text-4xl md:text-3xl uppercase animate__animated animate__fadeIn"
  >
    hola mundo
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/hola.jpg"
    />
  </div>
</Fragment>
`;

exports[`Probando el componente GifGrid Debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h2
    className="text-white text-center py-5 text-4xl md:text-3xl uppercase animate__animated animate__fadeIn"
  >
    hola mundo
  </h2>
  <div
    className="spinner"
  >
    <div
      className="cube1"
    />
    <div
      className="cube2"
    />
  </div>
  <div
    className="card-grid"
  />
</Fragment>
`;
